src/responses.rs

```diff
@@ -58,11 +76,7 @@ impl From<RawResponse> for Response {
                 let error = serde_json::from_str(&error).expect("JSON Deserialization");
                 ResponseType::Error(error)
             }
-            (None, Some(msg), None, None) => {
-                //Hack to remove extra escape characters
-                let msg = serde_json::from_str(&msg).expect("JSON Deserialization");
-                ResponseType::Subscribe(msg)
```

Comment:
With the suggestion above, we would only do the following patch here:

``` diff
-               //Hack to remove extra escape characters
-               let msg = serde_json::from_str(&msg).expect("JSON Deserialization");
                ResponseType::Subscribe(msg)
```

Reply:
> Thinking about this some more and running the failing tests myself, I think instead of serializing and deserializing a JSON structure in sdk-rust, we should leave it up to the user of sdk-rust what to send, and instead do the serialization and deserialization in the concrete test plans.
> 
> The benefit is, that a user can publish any format and is not bound to JSON.

Could you share the benefits you see?
My thinking might be "tainted" by the go-sdk but I think defaulting to JSON is a good thing:

* Test implementers don't have to think about serialization schemes,
* SDKs can automate encoding/decoding (the go SDK uses reflection to convert JSON payloads from/to Go structures),
* It's hard to create a JSON payload that cannot be used from another language
  * I'm seeing this now with the libp2p/ping go test, they publish a JSON we can use in Rust without much effort. If they had come up with their scheme, best case scenario we would use JSON, worst case we now have to support protocol buffer or a custom encoding.

For what it's worth, it looks like there is a type issue in the API that makes it confusing, but we don't have to leak this on the user side:
https://github.com/testground/sync-service/blob/d3e9e3e56403ed994f350942813d3d8aa746debc/types.go#L21
https://github.com/testground/sync-service/blob/d3e9e3e56403ed994f350942813d3d8aa746debc/types.go#L69
(string response)